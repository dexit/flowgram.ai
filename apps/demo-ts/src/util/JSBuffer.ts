export class JSBuffer {
  buffer: string[] = []

  constructor(public indent = '') {}

  clear(): JSBuffer {
    this.buffer = []
    return this
  }

  push(...args: string[]): JSBuffer {
    this.buffer.push(...args)
    return this
  }

  pushLine(...args: string[]): JSBuffer {
    this.buffer.push(...args, '\n')
    return this
  }

  toJs(separator: string = ''): string {
    // TODO js string perf
    // FIXME last line indent
    return (
      this.buffer
        .join(separator)
        .split('\n')
        .map((line) => `${this.indent}${line}`)
        // .map((line) => (line !== '\n' ? `${this.indent}${line}` : line))
        .join('\n')
    )
  }

  toString(): string {
    return this.toJs()
  }
}
