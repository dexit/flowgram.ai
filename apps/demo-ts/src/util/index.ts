export * from './log'
export * from './JSBuffer'
