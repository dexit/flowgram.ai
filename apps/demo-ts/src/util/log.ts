import debug from 'debug'

export const info = debug('log:info')
export const log = info
export const warn = debug('log:warn')
export const error = debug('log:error')
