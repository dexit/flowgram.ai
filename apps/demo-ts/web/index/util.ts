
export const codeMirrorBaseConfig: CodeMirror.EditorConfiguration = Object.freeze({
  mode: { name: 'javascript', json: true },
  lineNumbers: true,
  lineWrapping: false,
  theme: 'twilight',
  foldGutter: true,
  gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
  readOnly: true,
  cursorBlinkRate: -1,
})