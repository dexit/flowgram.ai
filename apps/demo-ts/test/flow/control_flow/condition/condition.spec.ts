import { runCases } from '../../util'

runCases(['condition_outputs'], __dirname)
