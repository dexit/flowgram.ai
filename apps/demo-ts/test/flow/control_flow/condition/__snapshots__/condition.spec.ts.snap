// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flow condition_outputs | {} 1`] = `
"export async function __main(__args) {
  //=== start@start | {level:1,blockIndex:0}
  const start = {}
  start.outputs = {
    score: 70,
  }
  Object.assign(start.outputs, __args)
  //=== switch_mode@condition | {level:1,blockIndex:1}
  const switch_mode = {}
  switch_mode.outputs = {
    rank: '',
  }
  //=== case_0@case | {level:1,blockIndex:0,parent:'switch_mode'}
  if (start.outputs.score >= 90) {
    //=== assign_0@variable | {level:2,blockIndex:0,parent:'case_0'}
    const assign_0 = {}
    assign_0.outputs = {}
    switch_mode.outputs.rank = 'Excellent'
  } // case_0@case
  //=== case_1@case | {level:1,blockIndex:1,parent:'switch_mode'}
  else if (start.outputs.score >= 60 && start.outputs.score < 90) {
    //=== assign_1@variable | {level:2,blockIndex:0,parent:'case_1'}
    const assign_1 = {}
    assign_1.outputs = {}
    switch_mode.outputs.rank = 'Normal'
  } // case_1@case
  //=== case_default@default | {level:1,blockIndex:2,parent:'switch_mode'}
  else {
    //=== assign_2@variable | {level:2,blockIndex:0,parent:'case_default'}
    const assign_2 = {}
    assign_2.outputs = {}
    switch_mode.outputs.rank = 'Fail'
  } // case_default@default
  //=== end@end | {level:1,blockIndex:2}
  const end = {}
  end.data = {
    rank: switch_mode.outputs.rank,
  }
  return end.data
}
"
`;

exports[`flow condition_outputs | {} 2`] = `
{
  "error": {
    "code": "",
    "msg": "",
  },
  "rank": "Normal",
}
`;
